codegen-client/src/test/kotlin/software/amazon/smithy/rust/codegen/client/smithy/customizations/serde/SerdeDecoratorTest.kt

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.rust.codegen.client.smithy.customizations.serde
+
+import org.junit.jupiter.api.Test
+import software.amazon.smithy.rust.codegen.client.testutil.clientIntegrationTest
+import software.amazon.smithy.rust.codegen.core.rustlang.CargoDependency
+import software.amazon.smithy.rust.codegen.core.rustlang.CratesIo
+import software.amazon.smithy.rust.codegen.core.rustlang.RustType
+import software.amazon.smithy.rust.codegen.core.rustlang.rustTemplate
+import software.amazon.smithy.rust.codegen.core.testutil.IntegrationTestParams
+import software.amazon.smithy.rust.codegen.core.testutil.asSmithyModel
+import software.amazon.smithy.rust.codegen.core.testutil.integrationTest
+import software.amazon.smithy.rust.codegen.core.testutil.unitTest
+
+/**
+ * The serde decorators for the client and the server are _identical_, but they live in separate Gradle projects.
+ * There's no point in testing everything twice, since the code is the same. Hence this file just contains a simple
+ * smoke test to ensure we don't disable the client decorator _somehow_; all tests testing the decorator's logic should
+ * live in `SerdeDecoratorTest` of the `codegen-server` project instead.
+ */
```

Comment:
this isn't really true. The code they generate is identical but client and server shapes are not the same—we really need to test against both.